File: Minimum_Number_Game/minimum_number_game.cpp
#include <vector>
#include <queue>
using namespace std;

//Non-optimized approach. I'm just doing problems that allow me to use Priority Queues to practice DSA's

class Solution {
    public:
        vector<int> numberGame(vector<int>& nums) {
            priority_queue<int, vector<int>, greater<int>> pq(nums.begin(), nums.end());
    
            vector<int> res;
            while (!pq.empty()) {
                int temp = pq.top();
                pq.pop();
                res.push_back(pq.top());
                pq.pop();
                res.push_back(temp);
    
            }
    
            return res;
            
        }
    };

File: Minimum_Number_Game/minimum_number_game.py
from queue import PriorityQueue
#Non-optimized approach. I'm just doing problems that allow me to use Priority Queues to practice DSA's
class Solution:
    def numberGame(self, nums: List[int]) -> List[int]:
        pq = PriorityQueue()
        res = []
        for i in nums:
            pq.put(i)

        while not pq.empty():
            temp = pq.get()
            res.append(pq.get())
            res.append(temp)

        return res